openjdk-testing/src/main/java/org/conscrypt/testing/TestClient.java

```diff
@@ -39,8 +37,8 @@ public TestClient(SSLSocket socket) {
     public void start() {
         try {
             socket.startHandshake();
-            input = new BufferedInputStream(socket.getInputStream());
-            output = new BufferedOutputStream(socket.getOutputStream());
+            input = socket.getInputStream();
```

Comment:
I think this might have been interfering with the `messageSize` selection which basically made them all the same. It also masked the issue that @bsidhom was seeing I believe.